lib/services/beanstalk/deployable-artifact.js

```diff
@@ -33,62 +35,120 @@ function uploadDeployableArtifactToS3(serviceContext, fileToUpload, fileExtensio
+    {
+      if(dat) winston.debug('Dockerrun.aws.json contents:','\n'+dat);
+      else    winston.debug('No Dockerrun.aws.json');
+      return dat;
+    })
+    .then(()=>
+    {
+      ebextensions.addEbextensionsToDir(ebextensionsToInject, tempDirPath);
+      return zipDir(tempDirPath)
+      .then(fileToUpload =>
+      {
+        let fileExtension = 'zip';
+        return uploadDeployableArtifactToS3(ownServiceContext, fileToUpload, fileExtension)
+        .then(s3ArtifactInfo =>
+        {
+          //Delete temp zip file
+          fs.unlinkSync(fileToUpload);
+          return s3ArtifactInfo;
         });
+      })
+      .then(s3ArtifactInfo =>
+      {
+        util.deleteFolderRecursive(tempDirPath); //Delete the whole temp folder
+        return s3ArtifactInfo;
+      });
+    });
 }
 
 //TODO - This is rather ugly code...
-function prepareAndUploadFile(ownServiceContext, pathToArtifact, ebextensionsToInject, fileName) {
-    let absArtifactPath = path.resolve(pathToArtifact);
-    let artifactDir = path.dirname(absArtifactPath);
-
-    let tempDirPath = makeTmpDir();
-    return util.copyFile(pathToArtifact, `${tempDirPath}/${fileName}`)
-        .then(() => {
-            let userEbextensionsDir = `${artifactDir}/.ebextensions/`;
-            if (fs.existsSync(userEbextensionsDir)) {
-                return util.copyDirectory(userEbextensionsDir, `${tempDirPath}/.ebextensions/`);
-            }
-            else {
-                return true;
-            }
-        })
-        .then(() => {
-            ebextensions.addEbextensionsToDir(ebextensionsToInject, tempDirPath)
-            return zipDir(tempDirPath)
-        })
-        .then(fileToUpload => {
-            let fileExtension = 'zip';
-            return uploadDeployableArtifactToS3(ownServiceContext, fileToUpload, fileExtension)
-                .then(s3ArtifactInfo => {
-                    //Delete temp zip file and temp dir
-                    fs.unlinkSync(fileToUpload);
-                    return s3ArtifactInfo;
-                });
-        })
-        .then(s3ArtifactInfo => {
-            util.deleteFolderRecursive(tempDirPath); //Delete the whole temp folder
-            return s3ArtifactInfo;
-        });
+function prepareAndUploadFile(accountConfig, ownServiceContext, pathToArtifact, ebextensionsToInject, fileName) {
+  let absArtifactPath = path.resolve(pathToArtifact);
+  let artifactDir = path.dirname(absArtifactPath);
+
+  let tempDirPath = makeTmpDir();
+  winston.debug('Copy to',tempDirPath,'file','\n'+JSON.stringify(pathToArtifact,null,2));
+  return util.copyFile(pathToArtifact, `${tempDirPath}/${fileName}`)
+  .then(()=>
+  {
+    let userEbextensionsDir = `${artifactDir}/.ebextensions/`;
+    if (fs.existsSync(userEbextensionsDir)) {
+      return util.copyDirectory(userEbextensionsDir, `${tempDirPath}/.ebextensions/`);
+    }
+    else {
+      return true;
+    }
+  })
+  .then(()=>
+  {
+    if(fileName == "Dockerrun.aws.json")
```

Comment:
Likewise, can we refactor a bit to use synchronous file calls here in these file changes too? That will make this a lot more readable too, and we have the luxury of doing that since this is a CLI tool that spends most of its time waiting for AWS.